### DISCUSSION TITLE
`rpm -qa` output in json

### DISCUSSION THREAD

**Discussion by kloczek** [rpm-software-management/rpm#2547] [General]
Someone have been trying to construct rpm query format string to produce for example `rpm -qa` packages names and versions as json? 🤔

**Comment by goeranu** [comment]
With a recent enough `rpm`, you don't need to. Rather, there is a `--json` option that give you json directly. (4.20 is new enough, I'm not sure if it is the first version with the option.)

**Reply by dmnks** [reply]
This is actually new in 4.20, yep.

**Reply by roke-julian-lockhart** [reply]
@dmnks, does an equivalent exist for `-qi`? It's damned close to YAML, but not valid.

**Reply by dmnks** [reply]
@RokeJulianLockhart not directly, but you could construct it yourself by using [rpm-queryformat(7)](https://rpm.org/docs/latest/man/rpm-queryformat.7) (see **Formats** specifically). In fact, both [`--info`](https://github.com/rpm-software-management/rpm/blob/294b68b872de851153fc882eee9d2970093dff4e/rpmpopt.in#L88) and [`--json`](https://github.com/rpm-software-management/rpm/blob/294b68b872de851153fc882eee9d2970093dff4e/rpmpopt.in#L122C13-L122C17) are implemented that way, too, as popt aliases. The latter basically just dumps all the available tags and wraps them in JSON, so you could do something similar by copying and adjusting the `--info` alias.
